src/orders/types.ts

```diff
@@ -1,8 +1,41 @@
 import {
+  OrderDetailsFragment,
+  OrderDetailsQuery,
+  OrderRefundDataQuery,
+} from "@dashboard/graphql";
+import {
+  OrderDetailsWithTransactionsFragment,
+  OrderDetailsWithTransactionsQuery,
   TransactionEventFragment,
   TransactionItemFragment,
 } from "@dashboard/graphql/transactions";
 
+export type ShopWithTransactions = OrderDetailsWithTransactionsQuery["shop"];
+export type ShopBothTypes = OrderDetailsQuery["shop"] | ShopWithTransactions;
+
+export type OrderBothTypes =
+  | OrderDetailsFragment
+  | OrderDetailsWithTransactionsFragment;
+
+/** use type from WithTransactions, exclude fields not available on old OrderDetails */
+export type OrderSharedType = Pick<
+  OrderDetailsWithTransactionsFragment,
+  keyof OrderDetailsFragment & keyof OrderDetailsWithTransactionsFragment
+>;
+
+export const isOrderWithTransactions = (
+  _order: any,
+  featureFlag: boolean,
+): _order is OrderDetailsWithTransactionsFragment => featureFlag;
```

Comment:
**question:** have there been problems specyfing order type? Also why is it underscore-prefixed?

Reply:
I've made this type guard to check if the order is a new type used when feature flag is enabled - I don't check anything inside the order, just if the featureFlag is set to enabled, this way we know that the order is using this specific type.